File: app/src/main/java/com/duy/pascal/backend/lib/graph/graphic_model/style/LineStyle.java
package com.duy.pascal.backend.lib.graph.graphic_model.style;

import android.graphics.DashPathEffect;

/**
 * Created by Duy on 08-Apr-17.
 */

public class LineStyle {
    //    draws a solid line.
    public static final int Solidln = 0;

    //    Draws a dotted line.
    public static final int Dottedln = 1;

    //    draws a non-broken centered line.
    public static final int Centerln = 3;

    //    draws a dashed line.
    public static final int Dashedln = 4;

    //2 - 2
    public static DashPathEffect dottedPath = new DashPathEffect(new float[]{10, 20}, 0);

    //3 - 2
    public static DashPathEffect ashedPath = new DashPathEffect(new float[]{10, 20}, 0);


}

File: libCompiler/src/main/assets/code_sample/graph/line_style.pas
Program line_style;
uses Graph;

var
    Gd, Gm: Integer;
    X1, Y1, X2, Y2: Integer;
begin
    Gd := Detect;
    InitGraph(Gd, Gm, ' ');
    if GraphResult <> grOk then
        Halt(1);

    SetColor(blue);

    //    draws a solid mLineNumber.
    SetLineStyle(SolidLn, 0, ThickWidth);
    rectangle(10, 10, 300, 110);

    //    Draws a dotted mLineNumber.
    SetLineStyle(DottedLn, 0, ThickWidth);
    rectangle(10, 120, 300, 220);

    //    draws a non-broken centered mLineNumber.
    SetLineStyle(CenterLn, 0, ThickWidth);
    rectangle(10, 230, 300, 330);

    //    draws a dashed mLineNumber.
    SetLineStyle(DashedLn, 0, ThickWidth);
    rectangle(10, 340, 300, 440);

    ReadLn;
    CloseGraph;
End.
